### DISCUSSION TITLE
How to inject rst in Python docstrings?

### DISCUSSION THREAD

**Discussion by wisp3rwind** [nvim-treesitter/nvim-treesitter#6358] [Q&A] [answered]
There are multiple issues concerned with the parsing of Python docstrings: https://github.com/nvim-treesitter/nvim-treesitter/pull/4199, https://github.com/nvim-treesitter/nvim-treesitter/pull/1240, https://github.com/nvim-treesitter/nvim-treesitter/issues/806, https://github.com/nvim-treesitter/nvim-treesitter/pull/1204, https://github.com/nvim-treesitter/nvim-treesitter/pull/5585, https://github.com/nvim-treesitter/nvim-treesitter/pull/917.

#1240 was closed with the comment "superseded by rewrite on main (and highlight/injection handling in upstream Neovim)", but from the documentation alone it's unclear to me how to achieve such injections. There's https://github.com/nvim-treesitter/nvim-treesitter/discussions/4655, but that seems to be concerned with just changing the associated highlight group, not adding an injection.

Thus: Is it currently possible to configure `nvim-treesitter` to inject rst into Python docstrings, and if so, how? Thanks!

**Comment by stsewd** [comment, marked as answer]
You can extend the injection query like this

https://github.com/stsewd/dotfiles/blob/ef7ee268a78456e2237f11211920daeacb7c2033/config/nvim/after/queries/python/injections.scm#L1-L4

**Reply by wisp3rwind** [reply]
Thanks! For reference, if someone comes across this discussion: It cannot be done directly from the lua config as prototyped in #1240, but one can simply drop rule files into the configuration directory. Namely, add 

```scm
; extends
; Docstrings
((expression_statement (string (string_content) @injection.content))
 (#set! injection.language "rst"))
```

as `.config/nvim/after/queries/python/injections.scm`.
